include/seqan3/io/structure/structure_file_format_dot_bracket.hpp

```diff
@@ -0,0 +1,399 @@
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides the seqan3::structure_file_format_dot_bracket class.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <charconv>
+#include <iterator>
+#include <stack>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+#include <range/v3/algorithm/copy.hpp>
+#include <range/v3/utility/iterator.hpp>
+#include <range/v3/view/chunk.hpp>
+#include <range/v3/view/drop_while.hpp>
+#include <range/v3/view/join.hpp>
+#include <range/v3/view/remove_if.hpp>
+#include <range/v3/view/transform.hpp>
+#include <range/v3/view/take_while.hpp>
+
+#include <seqan3/alphabet/nucleotide/rna5.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/io/detail/ignore_output_iterator.hpp>
+#include <seqan3/io/detail/output_iterator_conversion_adaptor.hpp>
+#include <seqan3/io/detail/misc.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/io/structure/detail.hpp>
+#include <seqan3/io/structure/structure_file_in_format_concept.hpp>
+#include <seqan3/io/structure/structure_file_out_format_concept.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/range/view/take_line.hpp>
+#include <seqan3/std/concept/range.hpp>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/transform.hpp>
+
+namespace seqan3
+{
+/*!\brief       The Dot Bracket format.
```

Comment:
Maybe we should call it Vienna instead? Because we use "dot bracket" already for an alphabet and it could be confusing. Maybe it's clearer to say that the STRUCTURE field is in "dot bracket", but the file is "vienna"? Also it would look nice when we add stockholm, because of city names 😄

Reply:
Vienna is a subset of this format, as Vienna doesn't allow for pseudo knots and therefore the alphabet must be limited to db3 instead of wuss51 and we have to drop energies. I'll think about it.

Reply:
I pushed a commit that switches the dot_bracket to vienna format.